src/ssl.c

```diff
@@ -14481,12 +14494,17 @@ void SetupSession(WOLFSSL* ssl)
 
     WOLFSSL_ENTER("SetupSession");
 
-    if (!IsAtLeastTLSv1_3(ssl->version) && ssl->arrays != NULL &&
-            !session->haveAltSessionID) {
+    if (!IsAtLeastTLSv1_3(ssl->version) && ssl->arrays != NULL) {
         /* Make sure the session ID is available when the user calls any
          * get_session API */
-        XMEMCPY(session->sessionID, ssl->arrays->sessionID, ID_LEN);
-        session->sessionIDSz = ssl->arrays->sessionIDSz;
+        if (!session->haveAltSessionID) {
+            XMEMCPY(session->sessionID, ssl->arrays->sessionID, ID_LEN);
+            session->sessionIDSz = ssl->arrays->sessionIDSz;
+        }
+        else {
+            XMEMCPY(session->sessionID, session->altSessionID, ID_LEN);
+            session->sessionIDSz = ID_LEN;
+        }
```

Comment:
I'm not sure what this change means. Can you explain the motivation?

Reply:
- When `ssl->arrays->sessionIDSz = 0` (no echo on blank ticket case), `session->altSessionID` is generated in the `createTicket()` function.
- This altSessionID is then used as `InternalTicket->id` for further processing.
- To ensure successful session retrieval from the cache, it is important to save this id value.